### ISSUE TITLE
"SyntaxError: encoding declaration in Unicode string" when vendorizing packages with PEP 263 unicode declaration

### ISSUE THREAD

**Issue by andy-maier** [mwilliamson/python-vendorize#13]
Sorry for bringing up a problem that only occurs on Python 2.7, but our package is in RHEL and they still support Python 2.7. Luckily, vendorize does not yet exclude Python 2.7!

Also, I found a simple solution (at the end of this description).

But first, here is the issue: When vendorizing a package that has files with a PEP 263 unicode declaration, the python-vendorize command fails after installing the vendorized packages when rewriting the imports:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/2.7.18/x64/bin/python-vendorize", line 8, in <module>
    sys.exit(main())
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/cli.py", line 10, in main
    vendorize_requirements(path=sys.argv[1])
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 20, in vendorize_requirements
    target_directory=target_directory,
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 30, in vendorize_requirement
    _rewrite_imports(target_directory, top_level_names)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 50, in _rewrite_imports
    _rewrite_imports_in_package(package_path, top_level_names, depth=1)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 59, in _rewrite_imports_in_package
    _rewrite_imports_in_package(child_path, top_level_names, depth=depth + 1)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 56, in _rewrite_imports_in_package
    _rewrite_imports_in_module(child_path, top_level_names, depth=depth)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/__init__.py", line 69, in _rewrite_imports_in_module
    rewritten_source = rewrite_imports_in_module(source, top_level_names, depth)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/site-packages/vendorize/import_rewrite.py", line 83, in rewrite_imports_in_module
    python_ast = ast.parse(source)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 0
SyntaxError: encoding declaration in Unicode string
```

The corresponding test run is https://github.com/pywbem/pywbem/runs/3064339874?check_suite_focus=true

That test run uses a version of python-vendorize that has debug code that prints the source code of the module that fails. 
The failing module is in the yamlloader package and it is the only one with a PEP263 unicode declaration:

File `yamlloader/ordereddict/__init__.py`:
```
# -*- coding: utf-8 -*-
"""YAML loaders and dumpers for PyYAML allowing to keep keys order."""
from __future__ import print_function, division, absolute_import

from .loaders import Loader, SafeLoader, CLoader, CSafeLoader
from .dumpers import Dumper, SafeDumper, CDumper, CSafeDumper

__all__ = ['CLoader',
           'Loader',
           'CDumper',
           'Dumper',
           'CSafeLoader',
           'SafeLoader',
           'CSafeDumper',
           'SafeDumper']
```

Removing that unicode declaration made the error go away. (See https://github.com/pywbem/pywbem/actions/runs/1029407801)

However, invoking `ast.parse()` on a copy of that file succeeds in the same Python 2.7 environment where vendorizing the package fails. See https://github.com/pywbem/pywbem/runs/3064615015?check_suite_focus=true, step "Debug - Test coding statements in ast.parse()". Also, while trying out different changes, the issue did not always surface on all Python 2.7 environments we are testing on (.e.g it showed up on Ubuntu but not on macOS or Windows).

The following SO articles are also related to this:
* https://stackoverflow.com/q/65294091/1424462 - compile() with unicode declaration on Python 2
* https://stackoverflow.com/a/39142445/1424462 - exec() with unicode declaration on Python 2

Since I don't think we can forbid that packages use unicode declarations, I am bringing this up here to discuss what the vendorize package could do about that.

I tried to solve this by encoding the source string to a UTF-8 binary string before invoking `ast.parse()`, when on Python 2.7, and this seems to work. See https://github.com/pywbem/pywbem/runs/3065193755?check_suite_focus=true.

I'll come forward with a PR shortly.

**Comment by andy-maier** [comment]
Addressed by PR #14.

**Comment by mwilliamson** [comment]
Closing since Python 2 is no longer supported.
